lexer/lexer_test.go

```diff
@@ -440,3 +442,30 @@ func TestCurrentPosition(t *testing.T) {
 		t.Fatalf("wrong position. expected=%d, got=%d", 6, l.CurrentPosition())
 	}
 }
+
+func TestUnicode(t *testing.T) {
+	input := `⺐ ❤ 😄`
```

Comment:
This is where I think we should be asking ourselves, do we want to support *all* unicode characters as identifiers? I think we don't have to, as doing so will create complications (the stuff I was talking about right above).

For example, node only seems to support letters:

```
 Welcome to Node.js v12.1.0.
Type ".help" for more information.
> ❤ = "aaa"
Thrown:
❤ = "aaa"


SyntaxError: Invalid or unexpected token
> 世界 = "aaa"
'aaa'
```

Go does the same:

* https://play.golang.org/p/7hEwxbeEJkz
* https://play.golang.org/p/qW83KxckYgF

So I think the example we should test is something like:

```
世界 = "⺐ ❤ 😄"
```

which means identifiers can be unicode letters, and strings can contain any unicode character.

Reply:
This is such a good point! 👍